### PR TITLE
[exporter/datadog] Update metrics export to use datadog-api-client-go instead of Zorkian

### LINKED ISSUE #6944: [exporter/datadog] add compression
Data sent from OpenTelemetry collector is not compressed and enabling gzip compression would improve bandwidth usage.

**Describe the solution you'd like**
Similar to what was done in this PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1747

### LINKED ISSUE #16776: [exporter/datadog] Migrate datadogexporter metric export to use native client APIs instead of Zorkian APIs
### Component(s)

exporter/datadog

### Is your feature request related to a problem? Please describe.

Currently Datadog exporter metrics export uses 3rd party APIs from `gopkg.in/zorkian/go-datadog-api.v2`. The Zorkian library is being deprecated.

### Describe the solution you'd like

Update count and gauge metric export to use the native client APIs from https://github.com/DataDog/datadog-api-client-go instead. This is guarded by feature gate and can be disabled by adding CLI flag `--feature-gates=-exporter.datadogexporter.metricexportnativeclient`.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by songy23**
**Description:** <Describe what has changed.>

Migrate metrics export in `datadogexporter` from `gopkg.in/zorkian/go-datadog-api.v2` to `github.com/DataDog/datadog-api-client-go/v2/api/datadog` because the Zorkian library is deprecated. This is guarded by feature gate and can be disabled by adding CLI flag --feature-gates=-exporter.datadogexporter.metricexportnativeclient

**Link to tracking Issue:** <Issue number if applicable>

Fixes #16776.
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6944.

**Testing:** <Describe what testing was performed and which tests were added.>

Tested in the staging environment, verified OTLP metrics are properly displayed on Datadog UI.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/datadogexporter/factory_test.go
Comment (mx-psi) [inline exporter/datadogexporter/factory_test.go:397]: Should we add a `defer <function that restores feature gate previous value>()` to tests? I am a bit worried that, since we are modifying global state, this will accidentally leak into other tests and we would be using a different metric export from the one we expect on a given test.

(Another option would be to pass the feature gate registry explicitly, but I think that would need more pervasive changes, so defer sounds like a reasonable compromise)

File: exporter/datadogexporter/internal/clientutil/api.go
Comment (mx-psi) [inline exporter/datadogexporter/internal/clientutil/api.go:36]: When would we get `endpoint == ""`? What should happen in that situation?

File: exporter/datadogexporter/internal/clientutil/api.go
Comment (mx-psi) [inline exporter/datadogexporter/internal/clientutil/api.go:69]: Why do we need this? If possible, we should ignore environment variables in the exporter code and leave that to the configuration resolver system (so here, in practical terms, never use `DD_API_Key` but `cfg.Api.Key` instead)

File: exporter/datadogexporter/internal/testutil/test_utils.go
Comment (mx-psi) [inline exporter/datadogexporter/internal/testutil/test_utils.go:48]: Could you add a comment justifying the `nosec` directive? (I am fine with it, but want to have the rationale recorded in-code)

File: exporter/datadogexporter/metrics_exporter.go
Comment (mx-psi) [inline exporter/datadogexporter/metrics_exporter.go:230]: I think we can just do
```suggestion
					_, _, merr := exp.metricsAPI.SubmitMetrics(ctx, datadogV2.MetricPayload{Series: ms})
```

if not using any optional parameters?

File: exporter/datadogexporter/internal/clientutil/api.go
Comment (mx-psi) [reply exporter/datadogexporter/internal/clientutil/api.go:35]: If left unspecified, the endpoint will be set based on the site https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e46477ef0700b313e7404d42b51f5bcab10273d2/exporter/datadogexporter/config.go#L491-L500, so I don't think we would get this in practice.

I have a slight preference to make the whole thing fail if the endpoint is empty, but since I don't feel strongly I will approve this and merge after you see this.

**Review by mx-psi — APPROVED** [review]

File: exporter/datadogexporter/internal/clientutil/api.go
Comment (songy23) [reply exporter/datadogexporter/internal/clientutil/api.go:35]: Sounds good, I'd prefer to do the validation in a follow-up PR since that's a fairly independent change.

**Comment by songy23** [comment]
Thanks for the review!

**Pull request merged**
